project/cmake/modules/FindFFMPEG.cmake

```diff
@@ -1,17 +1,199 @@
-include(ExternalProject)
-file(STRINGS ${CORE_SOURCE_DIR}/tools/depends/target/ffmpeg/FFMPEG-VERSION VER)
-string(REGEX MATCH "VERSION=[^ ]*$.*" FFMPEG_VER "${VER}")
-list(GET FFMPEG_VER 0 FFMPEG_VER)
-string(SUBSTRING "${FFMPEG_VER}" 8 -1 FFMPEG_VER)
-string(REGEX MATCH "BASE_URL=([^ ]*)" FFMPEG_BASE_URL "${VER}")
-list(GET FFMPEG_BASE_URL 0 FFMPEG_BASE_URL)
-string(SUBSTRING "${FFMPEG_BASE_URL}" 9 -1 FFMPEG_BASE_URL)
+# required ffmpeg library versions
+set(REQUIRED_FFMPEG_VERSION 3.1)
+set(_avcodec_ver ">=57.48.101")
+set(_avfilter_ver ">=6.47.100")
+set(_avformat_ver ">=57.41.100")
+set(_avutil_ver ">=55.28.100")
+set(_swscale_ver ">=4.1.100")
+set(_swresample_ver ">=2.1.100")
+set(_postproc_ver ">=54.0.100")
 
 
-if(ENABLE_INTERNAL_FFMPEG)
+# WARNING: this option is for developers as it will _disable ffmpeg version checks_!
+# Consider using FFMPEG_PATH instead, which _does_ check library versions
+#
+# Allows building with external ffmpeg not found in system paths,
+# without library version checks
+#
+# usage: -DWITH_FFMPEG=/path/to/ffmpeg_install_prefix
+if(WITH_FFMPEG)
+  set(FFMPEG_PATH ${WITH_FFMPEG})
+  message(STATUS "Warning: FFmpeg version checking disabled")
```

Comment:
I had that first, but don't really like the cmake Warning. It looks too much like an error.